components/brave_today/common/pref_names.cc

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_today/common/pref_names.h"
+
+#include "components/prefs/pref_service.h"
+
+namespace brave_news {
+bool GetIsEnabled(PrefService* prefs) {
+  bool should_show = prefs->GetBoolean(prefs::kNewTabPageShowToday);
```

Comment:
I think it'd be safer to redirect |prefs| to the original |prefs|. Previously, BrowserNewsControllerFactory::GetForContext() was doing the redirection but now that we have this standalone function, we might need that. 

```suggestion
  prefs = prefs->GetOriginalProfile();
  bool should_show = prefs->GetBoolean(prefs::kNewTabPageShowToday);
```

There could be more options for us here, but I"m not sure what is the best way.

* Add DCHECK() to prefs is not incognito?
* Or, GetIngognitoPersistentPrefsAllowlist? - this could be the most recommended way https://bravesoftware.slack.com/archives/C7VLGSR55/p1670000442736269

Reply:
Maybe add `DCHECK`? My initial concern with `GetOriginalProfile` is that if some UI in the browser calls this function to see if they should enable or show Brave News and that UI is also in the private window, then the feature will be active for the private profile, when it should not? Though I guess that only relies on the defaults being `false` which may always be that way.

Reply:
Maybe we should just return false in incognito windows?

```suggest
prefs = prefs.GetOriginalProfile();
...
return !Incognito() && should_show && opted_in
```

Reply:
Yeah, if brave news should be disabled on InCognito(), that would be better. In addition, if we should disable it on guest profiles too, then !IsOffTheRecord() would be even better.

And in favor of that, I think we should remove `BraveNewsControllerFactory::GetBrowserContextToUse` override at some point. This could misleading as if we could support Brave News on incognito.

Reply:
Hey, so is there an easy way to tell if you're `OffTheRecord` from components? There isn't a `GetOriginalProfile` method or any flags on `PrefService`.

Reply:
Oh, that's a good point. Could you try passing BrowserContext here and using BrowserContext::IsOffTheRecord?

Reply:
I assume we're fixing two issues in this PR:
1. BraveNews requests on a clean profile (a check added into `PublishersController::GetOrFetchPublishers` solves this).
2. BraveNews requests in incognito if it's enabled in the main profile (a `BrowserContext` passed everywhere solves this).

> And in favor of that, I think we should remove `BraveNewsControllerFactory::GetBrowserContextToUse` override at some point. This could misleading as if we could support Brave News on incognito.

I don't understand why are we passing `BrowserContext` to `GetIsEnabled` to disable `BraveNews` interaction in incognito, instead of just not creating `BraveNews`? Is there something that blocks us from doing so? If we can do this in this PR and keep things simple (nothing explodes, we don't need refactoring), then do it right here. If that's not the case, then we can fix (1) in this PR and do a separate PR to disable BraveNews in incognito entirely.

wdyt?

Reply:
Yup, I was wondering, removing `BraveNewsControllerFactory::GetBrowserContextToUse` could cause some code to access invalid nullptr. If scanning call sites and adding validity checks are not that heavy burden, removing it from this PR would be ideal.

Reply:
Seems to have been pretty painless to remove, to be honest. Thanks @goodov, this is definitely nicer.